### DISCUSSION TITLE
Bytearray Support

### DISCUSSION THREAD

**Discussion by jeffective** [wjakob/nanobind#554] [Q&A]
I am wrapping a library that does a lot of manipulation of c char arrays, like:

```c
char IOmap[4096];

int ecx_config_map_group(ecx_contextt *context, void *pIOmap, uint8 group)

```

so I was hoping to initialize this iomap bytearray in python and pass this into the c functions for it to be modified.

Is there plans to support bytearray in python? What alternatives do I have? I see ndarray but does this require my wrapper library to depend on numpy? I would like to avoid depending on numpy if I can get away with just a simple bytearray for now.

**Comment by hpkfft** [comment]
Yes, `nb::ndarray` supports Python's `bytearray`, and it does not depend on numpy (or on other modules).

The following function is in `test_ndarray.cpp` as a CI test:
```
NB_MODULE(test_ndarray_ext, m) {
    m.def("initialize", 
          [](nb::ndarray<unsigned char, nb::shape<10>, nb::device::cpu> &t) {
              for (size_t i = 0; i < 10; ++i)
                t(i) = (unsigned char) i;
          });
```
and is used in the python test code:
```
    import test_ndarray_ext as t
    a = bytearray(10)
    t.initialize(a)
    assert a == bytearray(range(10))
```
